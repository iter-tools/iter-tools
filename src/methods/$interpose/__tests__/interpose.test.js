/**
 * @generated-from ./$interpose.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

/* eslint-disable no-unused-vars,import/no-duplicates,no-constant-condition */

import { interpose } from '../../..';
import { wrap, unwrap } from '../../../test/helpers';

describe('interpose', () => {
  describe('when source is empty', () => {
    it('yields no values', () => {
      expect(unwrap(interpose('', null))).toEqual([]);
      expect(unwrap(interpose('', undefined))).toEqual([]);
      expect(unwrap(interpose('', wrap([])))).toEqual([]);
    });
  });

  describe('when source contains a single value', () => {
    it('yields that value', () => {
      const iter = interpose(null, wrap([1]));
      expect(unwrap(iter)).toEqual([1]);
    });
  });

  describe('when source contains multiple values', () => {
    it('yields interposed value between each value from source', () => {
      const iter = interpose(null, wrap([1, 2, 3]));
      expect(unwrap(iter)).toEqual([1, null, 2, null, 3]);
    });
  });

  describe('when source is a string', () => {
    it('warns', () => {
      interpose(null, 'abc');
      expect(console.warn).callsMatchSnapshot();
    });
  });
});
