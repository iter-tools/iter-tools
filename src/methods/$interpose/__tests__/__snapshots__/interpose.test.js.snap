// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interpose when source is a string warns 1`] = `
Array [
  Array [
    "For string inputs use interposeSeq instead of interpose",
  ],
]
`;
